### DISCUSSION TITLE
Dual infeasibility (inf_du) value increase after warm start

### DISCUSSION THREAD

**Discussion by Yunshan-Liu67** [coin-or/Ipopt#715] [Q&A]
Hi ,

I am currently using ipopt to solve a nonlinear problem in pyomo with [Model Scaling Transformation](https://pyomo.readthedocs.io/en/stable/model_transformations/scaling.html) and would like to utilize ipopt warm start function as well to have a faster convergence. For this propose, I followed the instruction [here](https://github.com/Pyomo/pyomo/blob/main/examples/pyomo/suffixes/ipopt_warmstart.py) and find out the inf_du did not drop after warm start as expected. The last iteration of the 1st run shows the following line in the log:
```
iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
 68  1.0044494e-01 9.96e-09 7.45e-09  -5.7 4.19e-05    -  1.00e+00 1.00e+00h  1
```
But the first iteration of the 2nd run shows the following:
```
iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
0  1.0044494e-01 7.59e-02 4.89e+07  -5.0 0.00e+00    -  0.00e+00 0.00e+00   0
```
As you can see, dual infeasibility (inf_du) is different at these two  iterations. Is it correct?
And the 2nd solve takes 33 iterations to solve, is it too much? how can we reduce this?

Many Thanks,
Yunshan

**Comment by svigerske** [comment]
You could increase the print_level to check whether the solution from the first solve also arrives as start point of the second solve and how much Ipopt modifies it. From the Pyomo code, it seems that you could try adding the settings
```
warm_start_bound_frac = 1e-6
warm_start_slack_bound_frac = 1e-6
warm_start_slack_bound_push = 1e-6
```
